be/src/storage/lake/tablet_manager.cpp

```diff
@@ -56,6 +60,50 @@ std::string TabletManager::txn_log_path(const std::string& group, const std::str
     }
 }
 
+std::string TabletManager::txn_log_cache_key(int64_t tablet_id, int64_t txn_id) {
+    return fmt::format("txn_{:016X}_{:016X}", tablet_id, txn_id);
+}
+
+static void tablet_meta_data_deleter(const CacheKey& key, void* value) {
+    std::string name = {key.data(), key.size()};
+    if (HasPrefixString(name, "tbl_")) {
+        TabletMetadataPtr* ptr = static_cast<TabletMetadataPtr*>(value);
+        delete ptr;
+    } else if (HasPrefixString(name, "txn_")) {
+        TxnLogPtr* ptr = static_cast<TxnLogPtr*>(value);
+        delete ptr;
+    }
+}
+
+bool TabletManager::put_into_metacache(const std::string& key_path, void* ptr, const int cost) {
+    CacheKey key = {key_path};
+    Cache::Handle* handle = _metacache->insert(key, ptr, cost, tablet_meta_data_deleter);
+    bool res = true;
+    if (handle == nullptr) {
+        res = false;
+    } else {
+        _metacache->release(handle);
+    }
+    return res;
+}
+
+void* TabletManager::get_from_metacache(const std::string& key_path) {
+    CacheKey key = {key_path};
+    Cache::Handle* handle = _metacache->lookup(key);
+    if (handle == nullptr) {
+        return nullptr;
+    } else {
+        auto ptr = _metacache->value(handle);
+        _metacache->release(handle);
+        return ptr;
+    }
+}
+
+void TabletManager::remove_from_metacache(const std::string& key_path) {
+    CacheKey key = {key_path};
```

Comment:
```suggestion
    CacheKey key(key_path);
```
https://starrocks.feishu.cn/docs/doccnH0eI3LRYUa40tsEZjviI3c#L0FL39